presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1160 @@
+        // Accumulo requires a column family and qualifier against a row ID, and these values
+        // are specified by the other columns in the table
+        if (meta.getColumns().size() == 1) {
+            throw new InvalidParameterException("Must have at least one non-row ID column");
+        }
+
+        // Check all the column types, and throw an exception if the types of a map are complex
+        // While it is a rare case, this is not supported by the Accumulo connector
+        ImmutableSet.Builder<String> columnNameBuilder = ImmutableSet.builder();
+        for (ColumnMetadata column : meta.getColumns()) {
+            if (Types.isMapType(column.getType())) {
+                if (Types.isMapType(Types.getKeyType(column.getType()))
+                        || Types.isMapType(Types.getValueType(column.getType()))
+                        || Types.isArrayType(Types.getKeyType(column.getType()))
+                        || Types.isArrayType(Types.getValueType(column.getType()))) {
+                    throw new PrestoException(VALIDATION,
+                            "Key/value types of a MAP column must be plain types");
+                }
+            }
+
+            columnNameBuilder.add(column.getName().toLowerCase());
+        }
+
+        ImmutableSet<String> columnNames = columnNameBuilder.build();
+
+        // Validate the columns are distinct
+        if (columnNames.size() != meta.getColumns().size()) {
+            throw new PrestoException(VALIDATION, "Duplicate column names are not supported");
+        }
+
+        // Column generation is for internal tables only
+        if (AccumuloTableProperties.getColumnMapping(meta.getProperties()) == null &&
+                AccumuloTableProperties.isExternal(meta.getProperties())) {
+            throw new PrestoException(VALIDATION,
+                    "Column generation for external tables is not supported, must specify " +
+                            AccumuloTableProperties.COLUMN_MAPPING);
+        }
+    }
+
+    /**
+     * Validates the columns set in the configure locality groups (if any) exist
+     *
+     * @param meta Table metadata
+     */
+    private void validateLocalityGroups(ConnectorTableMetadata meta)
+    {
+        // Validate any configured locality groups
+        Optional<Map<String, Set<String>>> groups =
+                AccumuloTableProperties.getLocalityGroups(meta.getProperties());
+        if (!groups.isPresent()) {
+            return;
+        }
+
+        String rowIdColumn = getRowIdColumn(meta);
+
+        // For each locality group
+        for (Map.Entry<String, Set<String>> g : groups.get().entrySet()) {
+            // For each column in the group
+            for (String col : g.getValue()) {
+                // If the column was not found, throw exception
+                List<ColumnMetadata> matched = meta.getColumns().stream()
+                        .filter(x -> x.getName().equalsIgnoreCase(col)).collect(Collectors.toList());
```

Comment:
Wouldn't it be better to reverse it: search `g.getValue()` for all values from `meta.getColumns()`?
This could be a bit more optimal when using `Set` implements `contains` in _O(lg(n))_. It would result in _O(n*lg(m))_ instead of _O(n*m)_, where _n_ is size of _meta.getColumns()_ and _m_ is size of _g.getValue()_.